pyproject.toml

```
@@ -24,7 +24,7 @@ dependencies = [
     "deprecated",
     "hats>=0.6.4",
     "nested-pandas>=0.4.7,<0.5.0",
-    "pyarrow>=14.0.1",
+    "pyarrow>=14.0.1,<21.0.0",
```

Do it for windows only?
```suggestion
    "pyarrow>=14.0.1,<21.0.0; platform_system == 'Windows'",
    "pyarrow>=14.0.1; platform_system != 'Windows'",
```